src/validator/receipt.ts

```diff
@@ -0,0 +1,85 @@
+import camelize, { type Camelize } from "camelize-ts";
+import {
+  type AsyncFunction,
+  type Signal,
+  type SignalAndInterval,
+  getAsyncPoller,
+} from "../helpers/await.js";
+import { type PartialRequired } from "../helpers/utils.js";
+import { hoistApiError } from "./errors.js";
+import {
+  type Components,
+  type FetchConfig,
+  type Paths,
+  getFetcher,
+  ApiError,
+} from "./client/index.js";
+
+export type Params =
+  Paths["/receipt/{chainId}/{transactionHash}"]["get"]["parameters"]["path"];
+
+type Response = Components["schemas"]["TransactionReceipt"];
+type AssertedResponse = PartialRequired<Response, "error_event_idx" | "error">;
+export type TransactionReceipt = Camelize<AssertedResponse>;
+
+function assertResponse(obj: Response): obj is AssertedResponse {
+  return (
+    obj.block_number != null &&
+    obj.chain_id != null &&
+    obj.transaction_hash != null &&
+    /* c8 ignore next */
+    (obj.table_id != null || obj.error != null || obj.error_event_idx != null)
+  );
+}
+
+function transformResponse(obj: Response): TransactionReceipt {
+  if (assertResponse(obj)) {
+    return camelize(obj);
+  }
+  /* c8 ignore next 2 */
+  throw new Error("malformed transaction receipt response");
+}
+
+export async function getTransactionReceipt(
+  config: FetchConfig,
+  params: Params,
+  opts: Signal = {}
+): Promise<TransactionReceipt> {
+  const receiptByTransactionHash = getFetcher(config)
+    .path("/receipt/{chainId}/{transactionHash}")
+    .method("get")
+    .create();
+  const { data } = await receiptByTransactionHash(params, opts).catch(
+    hoistApiError
+  );
+  const transformed = transformResponse(data);
+  return transformed;
+}
+
+export async function pollTransactionReceipt(
```

Comment:
@carsonfarmer We talked about this in the voice channel, but just putting a note here that the polling seems to keep happening for 6 iterations even if the materialization is confirmed on the first request.  Not exactly a critical bug, but it would be nice skip the unneeded http requests.

Reply:
Ooof, I totally forgot about this. I just pushed an update/fix and some tests to check for this. Thanks for reminding me!